src/number_theory/number_field/embeddings.lean

```diff
@@ -323,12 +323,132 @@ begin
     { rw ← complex_embedding.is_real_conjugate_iff at hφ,
       rwa ← h, }},
   { exact λ h, ⟨embedding w, h, mk_embedding w⟩, },
- end
+end
 
 lemma not_is_real_iff_is_complex {w : infinite_place K} :
   ¬ is_real w ↔ is_complex w :=
 by rw [is_complex_iff, is_real_iff]
 
+/-- For `w` a real infinite place, return the corresponding embedding as a morphism `K →+* ℝ`. -/
+noncomputable def is_real.embedding {w : infinite_place K} (hw : is_real w) : K →+* ℝ :=
+(is_real_iff.mp hw).embedding
+
+@[simp]
+lemma is_real.place_embedding_apply {w : infinite_place K} (hw : is_real w) (x : K):
+  place (is_real.embedding hw) x = w x :=
+begin
+  rw [is_real.embedding, complex_embedding.is_real.place_embedding, ← coe_mk],
+  exact congr_fun (congr_arg coe_fn (mk_embedding w)) x,
+end
+
+variable (K)
+
+/-- The map from real embeddings to real infinite places -/
+noncomputable def mk_real :
+  {φ : K →+* ℂ // complex_embedding.is_real φ} → {w : infinite_place K // is_real w} :=
+subtype.map mk (λ φ hφ, ⟨φ, hφ, rfl⟩)
+
+/-- The map from nonreal embeddings to complex infinite places -/
+noncomputable def mk_complex :
+  {φ : K →+* ℂ // ¬ complex_embedding.is_real φ} → {w : infinite_place K // is_complex w} :=
+subtype.map mk (λ φ hφ, ⟨φ, hφ, rfl⟩)
+
+/-- The map from real embeddings to real infinite places as an equiv -/
+noncomputable def mk_real_equiv :
```

Comment:
I think this one should be called `mk_real`, I don't see any reason to use the other one.

Reply:
Yeah, you're right. It is gone now ;)